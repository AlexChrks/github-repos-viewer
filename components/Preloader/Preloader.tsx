import React, {FC} from 'react'

import styles from './Preloader.module.css'

const Preloader: FC = () => {
  return (
    <div className={styles.preloaderWrapper}>
      <div className="preloader-wrapper small active">
      <div className="spinner-layer spinner-green-only">
        <div className="circle-clipper left">
          <div className="circle"></div>
        </div>
        <div className="gap-patch">
          <div className="circle"></div>
        </div>
        <div className="circle-clipper right">
          <div className="circle"></div>
        </div>
      </div>
    </div>
    </div>
  )
}

export default Preloader;
