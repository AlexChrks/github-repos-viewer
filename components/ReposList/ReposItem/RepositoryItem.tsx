import React from 'react'

interface RepositoryItemPropsInterface {
  repoName: string,
  updatedAt: string,
  starsCount: number,
  createdAt: string,
  size: number
  repoID: string,
  repoOwner: string
}

const RepositoryItem: React.FC<RepositoryItemPropsInterface> = ({ repoName, updatedAt, starsCount, createdAt, size }) => {

  return (
    <tr>
      <td>{repoName}</td>
      <td>{createdAt}</td>
      <td>{updatedAt}</td>
      <td>{starsCount}</td>
      <td>{size} kb</td>
    </tr>
  )
}

export default RepositoryItem
