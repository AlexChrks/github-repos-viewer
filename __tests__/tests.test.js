import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SignIn from '../pages/auth/sign-in';
import Preloader from '../components/Preloader/Preloader'
import ReposItem from '../components/ReposList/ReposItem/RepositoryItem'

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Sign In', () => {
  it('should render correctly', () => {
    const form = shallow(
      <SignIn />
    );
    expect(shallowToJson(form)).toMatchSnapshot();
  });
});


describe('Preloader', () => {
  it('should render correctly', () => {
    const preloader = shallow(
      <Preloader/>
    );
    expect(shallowToJson(preloader)).toMatchSnapshot();
  });
});

describe('RepositoryItem', () => {
  it('should render correctly', () => {
    const repoItem = shallow(
      <ReposItem/>
    );
    expect(shallowToJson(repoItem)).toMatchSnapshot();
  });
});
