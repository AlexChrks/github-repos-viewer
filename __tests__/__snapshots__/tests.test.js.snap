// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preloader should render correctly 1`] = `
<div
  className="preloaderWrapper"
>
  <div
    className="preloader-wrapper small active"
  >
    <div
      className="spinner-layer spinner-green-only"
    >
      <div
        className="circle-clipper left"
      >
        <div
          className="circle"
        />
      </div>
      <div
        className="gap-patch"
      >
        <div
          className="circle"
        />
      </div>
      <div
        className="circle-clipper right"
      >
        <div
          className="circle"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`RepositoryItem should render correctly 1`] = `
<tr>
  <td />
  <td />
  <td />
  <td />
  <td>
     kb
  </td>
</tr>
`;

exports[`Sign In should render correctly 1`] = `
<form
  className="signin_form"
  onSubmit={[Function]}
>
  <input
    name="githubToken"
    placeholder="Github token"
    type="text"
  />
  <button
    className="waves-effect waves-light btn"
    name="submitButton"
    type="submit"
  >
    Submit
  </button>
</form>
`;
